packages/codec/lib/format/types.ts

```diff
@@ -227,6 +268,9 @@ export interface ContractTypeForeign {
   kind: "foreign";
   typeName: string;
   contractKind?: Ast.ContractKind;
+  /**
+   * Indicates whether contract has payable fallback function
```

Comment:
Should also probably indicate why it is optional, or in what situations it won't be there

Reply:
I think the real answer to the latter is that it will always be there; I didn't realize that ABI was non-optional when I originally made this.  Doubtful I'd want to make this field non-optional, though, because it's a bit less essential?  IDK, maybe it should be non-optional, but it feels optional to me.

A better question should maybe be, should `contractKind` be optional given other changes?  That might always be there now.  Need to check.  That would be nice to make non-optional if possible.  Of course, that's not a documentation change.

Honestly I think this is fine as it is?  It's not like I'm explaining this for all the other fields.  Or maybe I could just put something like "left optional because it's not essential"?  IDK.

Reply:
Oh, I only made the point because I noticed a few places where you made optionality clear!

Reply:
Yeah, but those weren't in `Format`, were they? :)

Reply:
Actually though you have a point -- I should make a note somewhere explaining generally how optional fields in `Format` work (rather than addressing this one specifically).